### ISSUE TITLE
Implement reactions

### ISSUE THREAD

**Issue by towerofnix** [towerofnix/bantisocial#21]
The backend is *half* done already - there's an `/api/add-message-reaction` endpoint, and the reactions on a message can be fetched with `/api/message/:messageID` (or through message lists, etc). There still needs to be a socket event emitted to the client when a reaction is added, though. Input on how to do that? Perhaps just an "(add reaction `{username, messageID, reactionCharacter}` to message)" event? I don't want to send the whole list of everyone who's reacted every time - that could end up being a lot of data if there's many reactions (e.g. :+1: * 152, :-1: * 30, :100: * 4, :trollface: * 8).

The backend will also need to be updated to support more complex emoji, like ones which take up two unicode characters, and custom emotes, once those are implemented later (#18).

The client will need a way to add reactions. An emote picker would be handy, but that's for a separate issue - for now just entering a single character in a prompt is fine.

**Comment by heyitsmeuralex** [comment]
>Perhaps just an "(add reaction {username, messageID, reactionCharacter} to message)" event

Yes

**Comment by heyitsmeuralex** [comment]
@towerofnix can you stabilize the API for this - i.e. write the second half! - so I can work on the frontend for it?

**Comment by heyitsmeuralex** [comment]
Also kinda half blocked by #18.

**Comment by towerofnix** [comment]
Yeah, I'll see what I can do to make the format fairly extensible, so we can with custom reactions easily once we implement #18.

**Comment by towerofnix** [comment]
Unblocked; custom emotes are implemented.

**Comment by heyitsmeuralex** [comment]
This is gonna need docs too (`/api/add-message-reaction` was removed in #204). I'm thinking `POST /api/messages/:id/reactions`?
